src/_pytest/_code/code.py

```diff
@@ -1221,6 +1221,12 @@ def _write_entry_lines(self, tw: TerminalWriter) -> None:
         if not self.lines:
             return
 
+        if self.style == "value":
+            for line in self.lines:
+                tw.write(line)
```

Comment:
Using `TerminalWriter.write` instead of `TerminalWriter.line` because when using `TWMock` . we can't distinguish lines written with `TWMock._write_source` and lines written with `line` directly, whereas with `TWMock.write` we can, with the flag.

We could add a flag for `TWMock._write_source` as well, like so:
```python
class TWMock:
    WRITE = object()
    WRITE_SOURCE = object()

    def _write_source(self, lines, indents=()):
        if not indents:
            indents = [""] * len(lines)
        for indent, line in zip(indents, lines):
            newline = indent + line
            self.line((TWMock.WRITE_SOURCE, newline))
```

But then we'll have to change around half the test cases in `test_excinfo.py`

Reply:
Thanks for the explanation! I guess this rationale would be nice to have as a comment in the code itself, to help future readers.